core/src/ledger_metric_report_service.rs

```diff
@@ -0,0 +1,37 @@
+//! The `ledger_metric_report_service` periodically reports ledger store metrics.
+
+use {
+    solana_ledger::blockstore::Blockstore,
+    std::{
+        string::ToString,
+        sync::Arc,
+        thread::{self, Builder, JoinHandle},
+        time::Duration,
+    },
+};
+
+// Determines how often we report blockstore metrics.
+const BLOCKSTORE_METRICS_REPORT_PERIOD_MILLIS: u64 = 10000;
+
+pub struct LedgerMetricReportService {
+    t_cf_metric: JoinHandle<()>,
+}
+
+impl LedgerMetricReportService {
+    pub fn new(blockstore: Arc<Blockstore>) -> Self {
+        let t_cf_metric = Builder::new()
+            .name("metric_report_rocksdb_cf_metrics".to_string())
+            .spawn(move || loop {
+                thread::sleep(Duration::from_millis(
+                    BLOCKSTORE_METRICS_REPORT_PERIOD_MILLIS,
+                ));
+                blockstore.submit_rocksdb_cf_metrics_for_all_cfs();
```

Comment:
Hello, @joeaba. I would like to collect your feedback about this PR.

This PR creates a LedgerMetricReportService that reports RocksDB metrics for all column families periodically.
Currently, it is set to report for every 10 seconds. As every report will iterate through all column families, collect and report their RocksDB properties, we want to make sure the reporting period is frequent enough for us to collect useful information while minimizing its performance overhead.

If we want to minimize the overhead, the longest reasonable period in my opinion might be a minute.

Can I know your opinion about this?

Reply:
If it's influxdb performance you are concerned about, then don't worry it won't make a dent